maths/solovay_strassen_primality_test.py

```diff
@@ -0,0 +1,103 @@
+"""
+This script implements the Solovay-Strassen Primality test.
+
+This probabilistic primality test is based on Euler's criterion. It is similar
+to the Fermat test but uses quadratic residues. It can quickly identify
+composite numbers but may occasionally classify composite numbers as prime.
+
+More details and concepts about this can be found on:
+https://en.wikipedia.org/wiki/Solovay%E2%80%93Strassen_primality_test
+"""
+
+
+import random
+
+
+def jacobi_symbol(random_a: int, number: int) -> int:
+    """
+    Calculate the Jacobi symbol. The Jacobi symbol is a mathematical function
+    used to determine whether an integer is a quadratic residue modulo
+    another integer (usually prime) or not.
+
+    Parameters:
+        random_a (int): A randomly chosen integer from 2 to n-2 (inclusive)
+        number (int): The number that is tested for primality
+
+    Returns:
+        jacobi_symbol (int): The Jacobi symbol
```

Comment:
Please do not repeat the data types that are already checked by mypy in the function signature because readers will be confused if one is changed but the other is not.
```suggestion
        random_a: A randomly chosen integer from 2 to n-2 (inclusive)
        number: The number that is tested for primality

    Returns:
        jacobi_symbol: The Jacobi symbol
```
Repeating Jacobi symbol does not help the reader.  Perhaps the second one should be replaced with a definition of what a Jacobi symbol is.